registry/api/v2/urls.go

```diff
@@ -157,6 +157,18 @@ func (ub *URLBuilder) BuildBlobURL(ref reference.Canonical) (string, error) {
 	return layerURL.String(), nil
 }
 
+// BuildBlobRelativeURL constructs the relative url for the blob identified by name and dgst.
+func (ub *URLBuilder) BuildBlobRelativeURL(ref reference.Canonical) (string, error) {
```

Comment:
> The problem with URL builder is that cloned URLs always have http.ResolveReference called on them. Though I think this second approach is preferable.

Agreed. My first suggestion is :poop:. It is much simpler to configure the `URLBuilder` to not be relative, as their is a lot of code already trying to make it do the opposite.